### PR TITLE
Add missing methods for SamplingAlgorithm, fix docstrings

### PR DESCRIPTION
**Pull request by mryab**
Currently, `SamplingAlgorithm` inherits from `DecodingAlgorithm` but can't actually be used in its place due to missing `__call__` and unspecified `temperature` at the initialization time. This PR fixes this issue, as well as a couple of typos in docstrings.

I can also add `SamplingAlgorithm` to the list of classes allowed by `RemoteGenerationMixin._choose_sample_algorithm`: in essence, nothing should stop the user from choosing it at this point

### CODE REVIEW

**Comment by artek0chumak** [comment]
Hi! Thank you for PR. I have several questions.

File: src/petals/client/remote_generation.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
     DecodingAlgorithm,
     GreedyAlgorithm,
     NucleusAlgorithm,
+    SamplingAlgorithm,
```
Comment (artek0chumak) [inline src/petals/client/remote_generation.py:11]: Do we need this import?

File: src/petals/client/remote_generation.py [hunk 2]
```diff
@@ -101,14 +103,19 @@ def generate(
             elif num_beams is not None and num_beams > 1:
                 decoding_algorithm = BeamSearchAlgorithm(num_beams, batch_size=batch_size)
             else:
+                if top_k is not None or top_p is not None:
+                    logger.warning(
+                        "You passed top_k or top_p but did not set do_sample to True. Running greedy sampling"
```
Comment (borzunov) [inline src/petals/client/remote_generation.py:108]:
```suggestion
                        "You passed top_k or top_p but did pass do_sample=True. Running greedy sampling"
```

Comment (borzunov) [reply src/petals/client/remote_generation.py:108]: (makes more clear what a user should to type)

**Review by artek0chumak — APPROVED** [review]

File: src/petals/client/remote_generation.py [hunk 3]
```diff
@@ -102,14 +104,19 @@ def generate(
             elif num_beams is not None and num_beams > 1:
                 decoding_algorithm = BeamSearchAlgorithm(num_beams, batch_size=batch_size)
             else:
+                if top_k is not None or top_p is not None:
```
Comment (borzunov) [inline src/petals/client/remote_generation.py:107]:
```suggestion
                if temperature is not None or top_k is not None or top_p is not None:
```

Comment (borzunov) [inline src/petals/client/remote_generation.py:107]:
```suggestion
                if temperature is not None or top_k is not None or top_p is not None:
```

**Pull request merged**